packages/squiggle-lang/src/library/index.ts

```diff
@@ -18,38 +13,19 @@ const makeStdLib = (): Namespace => {
   res = res.merge(makeMathConstants());
   res = res.merge(makeVersionConstant());
 
-  // array and record lookups
+  // field lookups
   res = res.set(
     "$_atIndex_$",
     vLambda(
       new BuiltinLambda("$_atIndex_$", (inputs) => {
-        if (
-          inputs.length === 2 &&
-          inputs[0].type === "Array" &&
-          inputs[1].type === "Number"
-        ) {
-          const arr = inputs[0].value;
-          const index = inputs[1].value | 0; // TODO - fail on non-integer indices?
-          if (index >= 0 && index < arr.length) {
-            return arr[index];
-          } else {
-            return ErrorMessage.throw(
-              REArrayIndexNotFound("Array index not found", index)
-            );
-          }
-        } else if (
-          inputs.length === 2 &&
-          inputs[0].type === "Record" &&
-          inputs[1].type === "String"
-        ) {
-          const dict = inputs[0].value;
-          const index = inputs[1].value;
-          return (
-            dict.get(index) ??
-            ErrorMessage.throw(
-              RERecordPropertyNotFound("Record property not found", index)
-            )
-          );
```

Comment:
curious - what was going on here?

Reply:
This code was responsible for field lookups: `arr[3]`, `rec.foo` and `rec["foo"]`.

I needed to access `plotObj.fn` for rendering the function, so I had to make it more polymorphic and it made sense to move the code to `value.get` accessors on each value class.

Or, if you mean `??`, it's a https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing operator.

Reply:
I meant the former, thanks!